go/pkg/pass1/set-path.go

```diff
@@ -287,556 +222,497 @@ Partition:
 				}
 			}
 		}
-		unconnected = append(unconnected, zone1)
+		unconnectedPartitions = append(unconnectedPartitions, zone1)
 	}
+	return unconnectedPartitions
+}
 
-	// Check whether partitions are unconnected on purpose:
-	// Find zones with a partition-flag and partitions zone1.
-	var partitionZones []*zone
-	for _, z := range zones {
-		if z.partition != "" {
-			partitionZones = append(partitionZones, z)
-		}
-	}
-	var partitions_hash = make(map[*zone][]string)
-	var names = make(map[*zone]string)
-	for _, zone_i := range partitionZones {
-		var zone1 *zone
-		zone1 = findZone1(zone_i)
-		partitions_hash[zone1] = append(partitions_hash[zone1], zone_i.partition)
-		zone1.partition = zone_i.partition
-		names[zone1] = zone1.name
-	}
+func checkProperPartitionUsage(unconnectedPartitions []*zone) {
+
+	var partitions2PartitionTags = make(map[*zone][]string)
```

Comment:
Wozu generierst du eine frische Map, wenn sie in der nächsten Zeile gleich wieder überschrieben wird?